libs/block-mesh-manager/src/ws/connection_manager.rs

```diff
@@ -241,103 +236,66 @@ impl Broadcaster {
+                    }
+                    if let Some(new_messages) = settings.messages {
+                        messages = new_messages;
+                    }
+                    if let Some(new_window_size) = settings.window_size {
+                        window_size = new_window_size;
+                    }
                     let sent_messages_count = broadcaster
-                        .queue_multiple(messages.clone(), window_size)
+                        .queue_multiple(messages.clone(), window_size.clone())
                         .await;
                     if let Err(error) = stats_tx.send(CronReportStats::new(
-                        messages,
-                        window_size,
+                        messages.clone(),
+                        window_size.clone(),
                         sent_messages_count,
                     )) {
                         // TODO (send_if_modified, send_modify, or send_replace) can be used instead
                         tracing::error!("Could not sent stats, no watchers: {error}");
                     }
-                    tokio::time::sleep(period).await;
+                    tokio::time::sleep(period.clone()).await;
                 }
                 ()
             })
         };
 
-        let controller = CronReportsController::new(cron_task, channel_task, settings_tx, stats_rx);
+        let controller = CronReportsController::new(cron_task, stats_rx);
         self.cron_reports_controller = Some(controller.clone());
         controller
     }
 }
 
-#[tokio::test]
-async fn test_cron_reports() {
-    let mut conn_manager = ConnectionManager::new();
-    let (tx, mut rx) = mpsc::channel(10);
-    let user_id = Uuid::new_v4();
-    let addr = SocketAddr::from_str("127.0.0.1:8000").unwrap();
-    conn_manager.broadcaster.subscribe(user_id, addr, tx);
-    let mut controller = conn_manager.cron_reports(
-        Duration::from_secs(1),
-        vec![
-            WsServerMessage::RequestUptimeReport,
-            WsServerMessage::RequestBandwidthReport,
-        ],
-        10,
-    );
-    let msg = rx.recv().await.unwrap();
-    conn_manager.broadcaster.unsubscribe(user_id, addr);
-    let stats = controller.stats();
-    println!("{stats:?}");
-    tokio::time::sleep(Duration::from_secs(1)).await;
-    controller
-        .update(CronReportSettings {
-            period: None,
-            messages: None,
-            window_size: Some(10),
-        })
-        .await;
-    tokio::time::sleep(Duration::from_secs(1)).await;
-    let stats = controller.stats();
-    println!("{stats:?}");
+async fn fetch_latest_cron_settings(
+    pool: &PgPool,
+    user_id: Uuid,
+) -> anyhow::Result<CronReportSettings> {
+    let aggregate =
+        get_or_create_aggregate_by_user_and_name_pool(pool, AggregateName::CronReports, user_id)
+            .await?;
+    let settings: CronReportSettings = serde_json::from_value(aggregate.value)?;
+    Ok(settings)
```

Comment:
insert or update on table "aggregates" violates foreign key constraint "fk_user"

Reply:
No it doest if you use it as we said, it's not a random user but only a predefined "server user" , basically a constant.